src/main/scala/ai/privado/entrypoint/ScanProcessor.scala

```diff
@@ -201,10 +206,12 @@ object ScanProcessor extends CommandProcessor {
                             catLevelTwo = pathTree.apply(2),
                             categoryTree = pathTree,
                             language = Language.withNameWithDefault(pathTree.last),
-                            nodeType = NodeType.withNameWithDefault(pathTree.apply(3))
+                            nodeType = NodeType.withNameWithDefault(pathTree.apply(3)),
+                            isExternal = isExternal
                           )
                         )
-                        .filter(filterByLang)
+                        .filter(filterByLang),
+                      dedRules = configAndRules.dedRules
```

Comment:
why are we not doing rule validation and setting other fields as done for other parts